events/sender.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package events
+
+import (
+	"time"
+
+	log "github.com/cihub/seelog"
+)
+
+// RoleProvider indicates that event was sent from service provider
+const RoleProvider = "provider"
+
+// Sender builds events and sends them using given transport
+type Sender struct {
+	Transport Transport
+}
+
+// Transport allows sending events
+type Transport interface {
+	sendEvent(event) error
```

Comment:
By making this private you're forcing the transport to live in the same package. What if I actually want it to live in a sub package?

Reply:
Talking of which, might be a good idea to move the transports to a separate sub package, called transports

Reply:
But that's the idea - `Transport` is public, so you can manage `Transport` instances outside of package, deciding on the transport you want to use. But `Transport.sendEvent` shouldn't be used directly, so not exposing this method enforces users to use this as it is intended - only through `Sender`

Reply:
I would highly recommend designing metrics sending async right away. To make it as less intrusive as possible and not accidently hurt performance of any system components. A good problem example was - when consumer for any reason losses internet connection (DNS functionality to be exact) and tries to disconnect from node - disconnect hangs until final stats send timeouts - and thats not good.